js/data/bucket.js

```diff
@@ -327,16 +345,21 @@ Bucket.prototype.recalculateStyleLayers = function() {
     }
 };
 
-Bucket.prototype.populatePaintArrays = function(interfaceName, globalProperties, featureProperties, startGroup, startIndex) {
+Bucket.prototype.populatePaintArrays = function(interfaceName, globalProperties, featureProperties, startGroupIndex, startVertexIndex, endGroupIndex, endVertexIndex) {
     for (var l = 0; l < this.childLayers.length; l++) {
         var layer = this.childLayers[l];
         var groups = this.arrayGroups[interfaceName];
-        for (var g = startGroup.index; g < groups.length; g++) {
+        endGroupIndex = endGroupIndex || (groups.length - 1);
+
+        for (var g = startGroupIndex; g <= endGroupIndex; g++) {
             var group = groups[g];
             var length = group.layout.vertex.length;
             var vertexArray = group.paint[layer.id];
             vertexArray.resize(length);
 
+            var start = g === startGroupIndex ? startVertexIndex : 0;
+            var end = (endVertexIndex && g === endGroupIndex) ? endVertexIndex : length - 1;
```

Comment:
With this method now accepting ending indexes, maybe it would be simpler _require_ the end index params and move the responsibility for choosing `group.layout.vertex.length - 1` out to the callers?